### PR TITLE
Death to void params

### PR DESCRIPTION
**Pull request by AMurkin**
#### Summary
`SUMMARY: None`

#### Purpose of change
Based on [comment](https://github.com/CleverRaven/Cataclysm-DDA/pull/38519#pullrequestreview-368205682).

#### Describe the solution
Comment parameters names, remove void casts, rewrite some code.

#### Describe alternatives you've considered
None.

#### Testing
It compiles. Both tiles and curses.

#### Additional context
Did not touch `extern "C"` block in the src/sdltiles.cpp.
Not sure if `*/*param*/` is a problem. Github shows it in red in the src/activity_handlers.cpp.

### CODE REVIEW

File: src/animation.cpp [hunk 1]
```diff
@@ -437,13 +437,9 @@ void draw_bullet_curses( map &m, const tripoint &t, const char bullet, const tri
 #if defined(TILES)
 /* Bullet Animation -- Maybe change this to animate the ammo itself flying through the air?*/
 // need to have a version where there is no player defined, possibly. That way shrapnel works as intended
-void game::draw_bullet( const tripoint &t, const int i, const std::vector<tripoint> &trajectory,
+void game::draw_bullet( const tripoint &t, const int /*i*/, const std::vector<tripoint> &/*trajectory*/,
                         const char bullet )
```
Comment (kevingranade) [inline src/animation.cpp]:
```suggestion
void game::draw_bullet( const tripoint &t, const int /*i*/,
                        const std::vector<tripoint> &/*trajectory*/, const char bullet )
```

File: src/wish.cpp [hunk 2]
```diff
@@ -319,11 +319,7 @@ class wish_monster_callback: public uilist_callback
             wrefresh( w_info );
         }
 
-        bool key( const input_context &, const input_event &event, int entnum, uilist *menu ) override {
-            // Unused
-            ( void )entnum;
-            // Unused
-            ( void )menu;
+        bool key( const input_context &, const input_event &event, int /*entnum*/, uilist */*menu*/ ) override {
```
Comment (kevingranade) [inline src/wish.cpp]:
```suggestion
        bool key( const input_context &, const input_event &event, int /*entnum*/,
                  uilist */*menu*/ ) override {
```

**Comment by ZhilkinSerg** [comment]
> Not sure if */*param*/ is a problem. Github shows it in red in the src/activity_handlers.cpp.

VS complains about it too (`*/` resembles closing of multiline comment, but without opening `/*`). I think warning can be avoided with additional whitespace after first asterisk (i.e. `* /*param*/).

**Comment by jkraybill** [comment]
Yeah this is the error from VS:
warning C4138: '*/' found outside of comment

Would be nice to not have this, but it's not a dealbreaker!

**Pull request merged**